identity-apps-core/apps/recovery-portal/src/main/webapp/includes/language-switcher.jsp

```diff
@@ -65,26 +65,30 @@
      * @param name - Name of the cookie
      * @param value - Value to be stored
      * @param days - Expiry days
+     * @param options - Additional options for the cookie such as `httpOnly` and `secure.
      */
-    function setCookie(name, value, days) {
+    function setCookie(name, value, days, options) {
         var expires = "";
         var domain = ";domain=" + URLUtils.getDomain(window.location.href);
 
         if (days) {
             const date = new Date();
-            date.setTime(date.getTime() + (days*24*60*60*1000));
+            date.setTime(date.getTime() + (days * 24 * 60 * 60 * 1000));
             expires = "; expires=" + date.toUTCString();
         }
 
-        document.cookie = name + "=" + (value || "")  + expires + domain + "; path=/";
+        var httpOnlyString = (options && options.httpOnly) ? "; HttpOnly" : "";
+        var secureString = (options && options.secure) ? "; Secure" : "";
+
+        document.cookie = name + "=" + (value || "") + expires + domain + "; path=/" + httpOnlyString + secureString;
     }
 
     /**
      * Handles language change by setting the `ui_locale` cookie.
      */
     function setUILocaleCookie(language) {
         var EXPIRY_DAYS = 30;
-        setCookie('ui_lang', language, EXPIRY_DAYS);
+        setCookie('ui_lang', language, EXPIRY_DAYS, { secure: true });
```

Comment:
if we set ui_lang to secure, js logic will fail to read the cookie, won't it?

https://github.com/brionmario/identity-apps/blob/master/identity-apps-core/apps/recovery-portal/src/main/webapp/includes/language-switcher.jsp#L43

Reply:
I think what you mean are the `httpOnly` cookies, `secure` will only restrict the cookie to be sent over an HTTPS connection. We can still access the cookie from JS.

Reply:
Ah yes, my bad :D